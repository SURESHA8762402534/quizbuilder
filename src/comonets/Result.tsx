import { Paper, Typography } from "@mui/material";
import { useEffect, useState } from "react";
import { useLocation } from "react-router-dom"

const Result = ()=>{
    const data:any = useLocation().state;
    
    const rightAns = Array.from(data)?.filter((item:any)=>{
        return JSON.stringify(item.ans) === JSON.stringify(Array.from(item.userAns).sort())
    })

  const  score = rightAns.length;


    return(
        <>
        <Typography variant="h4" sx={{m:3,textAlign:'center',textDecoration:'underline'}}>RESULT</Typography>
        <Paper>
            <Paper sx={{float:'left',p:3}}>
                <Paper sx={{width:'20vw',p:2,textAlign:'center', backgroundColor:'lightcyan'}}>
                    <Typography>
                        Total Score is <b>{score}/5</b>
                    </Typography>
                    <Typography>
                        Total percentage is <b>{((score*100)/5).toFixed(2)}%</b>
                    </Typography>
                </Paper>
                <Paper sx={{mt:3,ml:2,backgroundColor:'whitesmoke',p:2}}>
                    {Array.from(data)?.map((item:any,idx:any)=>{
                        return (
                            item.type !== 'multi'?
                            <>
                            <Typography variant="h6">
                                Q{item.id}. {item.que}
                            </Typography>
                            <Typography sx={{backgroundColor:'lightgreen',m:1,p:1}}>
                                Correct Answer is --- {[...item.ans]}
                            </Typography>
                            <Typography sx={{backgroundColor:'lightyellow',m:1,p:1}}>
                                Your Answer is --- {[...item.userAns]}
                            </Typography>
                            </>
                            :
                            <>
                             <Typography variant="h6">
                                Q{item.id}. {item.que}
                            </Typography>
                            <Typography sx={{backgroundColor:'lightgreen',m:1,p:1}}>
                                Correct Answer is --- {item.ans[0]} , {item.ans[1]}
                            </Typography>
                            <Typography sx={{backgroundColor:'lightyellow',m:1,p:1}}>
                                Your Answer is --- {item.userAns[0]} ,  {item.userAns[1]} 
                            </Typography>
                            </>
                        )
                    })}
                </Paper>

            </Paper>
            
            <Paper sx={{float:'right',p:3,mr:20,mt:5}}>
                <div data-testid='piechart' style={{
                    width:'20rem',
                    height:'20rem',
                    borderRadius:'50%',
                    position:'sticky',
                    top:'10px',
                    backgroundImage:`conic-gradient(green 0deg, green ${score * 75}deg, red ${score * 75}deg, red 360deg)`
                }}>
                    <Typography variant="h4" sx={{color:'white',textAlign:'right',mr:'2rem',pt:20,pr:1}}>
                    <b>{((score*100)/5).toFixed(2)}%</b>
                    </Typography>
                </div>
            </Paper>
        </Paper>
        </>
    )
}

export default Result