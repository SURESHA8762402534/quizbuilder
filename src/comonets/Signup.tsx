import { Button, FormControlLabel, Paper, Radio, RadioGroup, TextField, Typography } from "@mui/material";
import { useState } from "react"
import { useNavigate } from "react-router-dom";

const Signup = ()=>{
    const [input,setInput] = useState('');
    const naviagte = useNavigate();
    const [name, setName] = useState('');
    const [gender, setgender] = useState('')

    const handelSubmit = (e:any)=>{
        console.log(input);
        e.preventDefault();
        naviagte('/quiz',{state:input})
    }

    return(
        <>
        <Paper elevation={12} sx={{width:'30vw',ml:'35vw',mt:3,p:2,backgroundColor:'blanchedalmond'}}>
            <form action="" onSubmit={handelSubmit}>
                <Typography variant="h4" sx={{textAlign:'center',m:3}}>Signup</Typography>

                <TextField placeholder="enter your name" value={name} onChange={(e:any)=>setName(e.target.value)}/>
                <Typography sx={{mt:2,mb:1}}>Select gender</Typography>
                <RadioGroup  row sx={{m:2}} onClick={(e:any)=>setgender(e.target.value)}>
                    <FormControlLabel data-testid='gender1' control={<Radio/>} label='male' value='male'/>
                    <FormControlLabel data-testid='gender2' control={<Radio/>} label='female' value='female'/>
                    <FormControlLabel data-testid='gender3' control={<Radio/>} label='other' value='other'/>
                </RadioGroup>
                <Typography sx={{mt:2,mb:1}} variant='h6'>Select language</Typography>
                <RadioGroup onClick={(e:any)=>setInput(e.target.value)} sx={{m:2}}>
                    <FormControlLabel data-testid='english' control={<Radio/>} label='english' value='english'/>
                    <FormControlLabel data-testid='hindi' control={<Radio/>} label='hindi' value='hindi'/>
                   
                </RadioGroup>
                <Typography sx={{textAlign:'center'}}>
                    <Button disabled={!input || !gender || !name} variant='contained' type='submit'>Start Test</Button>
                </Typography>
            </form>
        </Paper>
        </>
    )
}
export default Signup