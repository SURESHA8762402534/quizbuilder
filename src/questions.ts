export const questions = [
    {
        id:1,
        type:'mcq',
        que:'what is React?',
        opt:[
            'librery',
            'framework',
            'both',
            'none'
        ],
        ans:['librery'],
        userAns:[],
        isAnsd:false
    },
   
    {
        id:2,
        type:'bool',
        que:'React is a librery',
        opt:[
            'true',
            'false'
            
        ],
        ans:['true'],
        userAns:[],
        isAnsd:false
    },
    {
        id:3,
        type:'match',
        que:'match the following',
        match:[
            '1)React - a)Framework',
            '2)Angular - b)state management tool',
            '3)Redux - c)Librery'
            
        ],
        opt:[
            '1-a,2-b,3-c',
            '1-c,2-b,3-a',
            '1-c,2-a,3-b',
            'none'
        ],
        ans:['1-c,2-a,3-b'],
        userAns:[],
        isAnsd:false
    },
    {
        id:4,
        type:'multi',
        que:'React',
        opt:[
            'librery',
            'framework',
            'uses jsx',
            'none'
        ],
        ans:['librery','uses jsx'],
        userAns:[],
        isAnsd:false
    },
    {
        id:5,
        type:'fill',
        que:'React is built by ____',
       
        ans:['facebook'],
        userAns:[],
        isAnsd:false
    },
]