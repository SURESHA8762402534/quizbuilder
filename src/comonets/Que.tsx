import { ArrowBack, ArrowForward } from "@mui/icons-material";
import {
  Button,
  Checkbox,
  FormControlLabel,
  Grid,
  Paper,
  Radio,
  RadioGroup,
  TextField,
  Typography,
} from "@mui/material";
import { useEffect, useState } from "react";
import { useLocation, useNavigate } from "react-router-dom";
import { questions } from "../questions";

const Que = () => {
  const language = useLocation().state;
  const [data, setData]: any = useState();
  const [page, setPage] = useState(0);
  const [input, setInput] = useState("");
  const [count, setCont] = useState(0)

  const navigate = useNavigate();

  const fetchData = () => {
    if (language === "english") {
      setData(questions);
    } else {
      setData(questions);
    }
  };

  useEffect(() => {
    fetchData();
  }, []);

  const handelInput = (e: any) => {
    console.log(e.target.value);
    setCont((p:any)=>p+1)
    data[page].isAnsd = true;
    setInput(e.target.value);
    data[page].userAns.splice(0,1)
    data[page].userAns.push(e.target.value);
    console.log(data[page]);
  };

  const handelArr = (e: any) => {
    data[page].isAnsd = true;
    if (e.target.value !== undefined) {
      setCont((p:any)=>p+1)
      if (data[page].userAns.length < 2) {
        data[page].userAns.push(e.target.value);
      } else {
        data[page].userAns.splice(0, 1);
        data[page].userAns.push(e.target.value);
      }
    }
    console.log(data[page]);
  };

  const displayOpt = () => {
    switch (data[page]?.type) {
      case "mcq":
        return (
          <>
            {data[page]?.opt.map((item: any, idx: any) => {
              return (
                <RadioGroup
                  key={idx}
                  onClick={handelInput}
                  // defaultValue={data[page]?.userAns}
                >
                  <FormControlLabel
                    data-testid={`mcq-${idx}`}
                    control={<Radio checked={data[page]?.userAns.includes(item)} />}
                    value={item}
                    label={item}
                  />
                </RadioGroup>
              );
            })}
          </>
        );
      case "fill":
        return (
          <>
            <TextField
              data-testid="fill"
              value={data[page]?.userAns}
              placeholder="enter your answer here"
              onChange={handelInput}
            />
          </>
        );
      case "bool":
        return (
          <>
            {data[page]?.opt.map((item: any, idx: any) => {
              return (
                <RadioGroup
                  key={idx}
                  onClick={handelInput}
                  // defaultValue={data[page]?.userAns}
                >
                  <FormControlLabel
                    data-testid={`bool-${idx}`}
                    control={<Radio checked={data[page]?.userAns.includes(item)}/>}
                    value={item}
                    label={item}
                  />
                </RadioGroup>
              );
            })}
          </>
        );
      case "match":
        return (
          <>
            <Grid>
              <Grid item>
                {data[page]?.match.map((item: any, idx: any) => {
                  return <Typography sx={{ m: 2 }}>{item}</Typography>;
                })}
              </Grid>
              <Grid item>
                {data[page]?.opt.map((item: any, idx: any) => {
                  return (
                    <RadioGroup
                      key={idx}
                      onClick={handelInput}
                      // defaultValue={data[page]?.userAns}
                    >
                      <FormControlLabel
                        data-testid={`match-${idx}`}
                        control={
                          <Radio checked={data[page]?.userAns.includes(item)} />
                        }
                        value={item}
                        label={item}
                      />
                    </RadioGroup>
                  );
                })}
              </Grid>
            </Grid>
          </>
        );
      case "multi":
        return (
          <>
            {data[page]?.opt.map((item: any, idx: any) => {
              return (
                <RadioGroup
                  key={idx}
                  onClick={handelArr}
                  // defaultValue={data[page]?.userAns}
                >
                  <FormControlLabel
                    data-testid={`multi-${idx}`}
                    control={
                      <Checkbox
                      //  checked={data[page]?.userAns.includes(item)}
                      />
                    }
                    value={item}
                    label={item}
                  />
                </RadioGroup>
              );
            })}
          </>
        );
    }
  };

  return (
    <>
      {data ? (
        <>
          <Typography sx={{ textAlign: "center", mb: 3 }}>
            {data?.map((item: any, idx: any) => {
              return (
                <Button
                  key={idx}
                  data-testid={`btn-${idx}`}
                  variant="outlined"
                  style={
                    item.isAnsd
                      ? { backgroundColor: "red", color: "white" }
                      : { backgroundColor: "gray", color: "white" }
                  }
                  sx={{ m: 2 }}
                  onClick={(e: any) => setPage(Number(e.target.innerText) - 1)}
                >
                  {item.id}
                </Button>
              );
            })}
          </Typography>
          <Paper
            elevation={16}
            sx={{ width: "30vw", ml: "32vw", mt: 4, mb: 4, p: 3 }}
          >
            <Typography variant="h6" sx={{ m: 1 }}>
              Q{data[page]?.id}. {data[page]?.que}
            </Typography>
            {displayOpt()}
          </Paper>

          <Grid container sx={{ mt: 4, ml: 3, mr: 3 }}>
            <Grid item xs={1}>
              <Button
                data-testid="back"
                variant="outlined"
                disabled={data[page]?.id <= 1}
                onClick={() => setPage((p: any) => p - 1)}
              >
                <ArrowBack />
              </Button>
            </Grid>
            <Grid item xs={8}></Grid>
            <Grid item xs={1}>
              <Button
                data-testid="submit"
                variant="outlined"
                sx={count<6 ? {display:'none'}:{display:'flex'}}
                onClick={() => navigate("/result", { state: data })}
              >
                Submit
              </Button>
            </Grid>
            <Grid item xs={1}>
              <Button
                data-testid="frd"
                variant="outlined"
                disabled={data[page]?.id >= data?.length}
                onClick={() => setPage((p: any) => p + 1)}
              >
                <ArrowForward />
              </Button>
            </Grid>
          </Grid>
        </>
      ) : null}
    </>
  );
};

export default Que;
