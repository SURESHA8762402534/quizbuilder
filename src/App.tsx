import React from 'react';
import logo from './logo.svg';
import './App.css';
import { Route, Routes } from 'react-router-dom';
import Signup from './comonets/Signup';
import Result from './comonets/Result';
import Que from './comonets/Que';

function App() {
  return (
   <>
   <Routes>
     <Route path='/' element={<Signup/>}/>
     <Route path='/quiz' element={<Que/>}/>
     <Route path='/result' element={<Result/>}/>
   </Routes>
   
   </>
  );
}

export default App;
