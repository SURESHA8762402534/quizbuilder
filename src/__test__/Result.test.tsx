import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../App";
import Que from "../comonets/Que";
import Result from "../comonets/Result";
import Signup from "../comonets/Signup";
import { questions } from "../questions";

jest.mock("react-router-dom",()=>{
    return{
        
        useLocation:()=>{
            return{
                state:[
                    {
                        id:1,
                        type:'mcq',
                        que:'what is React?',
                        opt:[
                            'librery',
                            'framework',
                            'both',
                            'none'
                        ],
                        ans:'librery',
                        userAns:'',
                        isAnsd:false
                    },
                    {
                        id:2,
                        type:'fill',
                        que:'React is built by ____',
                       
                        ans:'facebook',
                        userAns:'',
                        isAnsd:false
                    },
                    {
                        id:3,
                        type:'bool',
                        que:'React is a librery',
                        opt:[
                            'true',
                            'false'
                            
                        ],
                        ans:'true',
                        userAns:'',
                        isAnsd:false
                    },
                    {
                        id:4,
                        type:'match',
                        que:'match the following',
                        match:[
                            '1)React - a)Framework',
                            '2)Angular - b)state management tool',
                            '3)Redux - c)Librery'
                            
                        ],
                        opt:[
                            '1-a,2-b,3-c',
                            '1-c,2-b,3-a',
                            '1-c,2-a,3-b',
                            'none'
                        ],
                        ans:'1-c,2-a,3-b',
                        userAns:'',
                        isAnsd:false
                    },
                    {
                        id:5,
                        type:'multi',
                        que:'React',
                        opt:[
                            'librery',
                            'framework',
                            'uses jsx',
                            'none'
                        ],
                        ans:['librery','uses jsx'],
                        userAns:[],
                        isAnsd:false
                    },
                ]
            }
        }
    }
})

describe('result component test',()=>{
    test('snap shot',()=>{
        const tree = render(
            <Result/>
        );
        expect(tree).toMatchSnapshot();
        screen.debug()
    });
    test('test for componet',()=>{
        const tree = render(
            <Result/>
        );

        expect(screen.getByTestId('piechart')).toBeInTheDocument()
        expect(screen.getByTestId('piechart')).toHaveStyle('borderRadius:50%')
       
    });
    
})