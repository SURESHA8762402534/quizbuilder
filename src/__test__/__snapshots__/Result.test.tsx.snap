// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`result component test snap shot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-oak7j8-MuiTypography-root"
      >
        RESULT
      </h4>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1caak8a-MuiPaper-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1sek5l-MuiPaper-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Total Score is 
            <b>
              0
              /5
            </b>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Total percentage is 
            <b>
              0.00
              %
            </b>
          </p>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1kol39z-MuiPaper-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Q
            1
            . 
            what is React?
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
          >
            Correct Answer is --- 
            l
            i
            b
            r
            e
            r
            y
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
          >
            Your Answer is --- 
          </p>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Q
            2
            . 
            React is built by ____
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
          >
            Correct Answer is --- 
            f
            a
            c
            e
            b
            o
            o
            k
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
          >
            Your Answer is --- 
          </p>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Q
            3
            . 
            React is a librery
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
          >
            Correct Answer is --- 
            t
            r
            u
            e
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
          >
            Your Answer is --- 
          </p>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Q
            4
            . 
            match the following
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
          >
            Correct Answer is --- 
            1
            -
            c
            ,
            2
            -
            a
            ,
            3
            -
            b
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
          >
            Your Answer is --- 
          </p>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Q
            5
            . 
            React
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
          >
            Correct Answer is --- 
            librery
             , 
            uses jsx
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
          >
            Your Answer is --- 
             ,  
          </p>
        </div>
      </div>
      <div
        data-testid="piechart"
        style="width: 20rem; height: 20rem; border-radius: 50%; position: fixed; top: 100px; right: 10vw;"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-pfhr9p-MuiTypography-root"
        >
          <b>
            0.00
            %
          </b>
        </h4>
      </div>
    </div>
  </body>,
  "container": <div>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-oak7j8-MuiTypography-root"
    >
      RESULT
    </h4>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1caak8a-MuiPaper-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1sek5l-MuiPaper-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Total Score is 
          <b>
            0
            /5
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Total percentage is 
          <b>
            0.00
            %
          </b>
        </p>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1kol39z-MuiPaper-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Q
          1
          . 
          what is React?
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
        >
          Correct Answer is --- 
          l
          i
          b
          r
          e
          r
          y
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
        >
          Your Answer is --- 
        </p>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Q
          2
          . 
          React is built by ____
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
        >
          Correct Answer is --- 
          f
          a
          c
          e
          b
          o
          o
          k
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
        >
          Your Answer is --- 
        </p>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Q
          3
          . 
          React is a librery
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
        >
          Correct Answer is --- 
          t
          r
          u
          e
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
        >
          Your Answer is --- 
        </p>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Q
          4
          . 
          match the following
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
        >
          Correct Answer is --- 
          1
          -
          c
          ,
          2
          -
          a
          ,
          3
          -
          b
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
        >
          Your Answer is --- 
        </p>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Q
          5
          . 
          React
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hy60wk-MuiTypography-root"
        >
          Correct Answer is --- 
          librery
           , 
          uses jsx
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-qt3w6v-MuiTypography-root"
        >
          Your Answer is --- 
           ,  
        </p>
      </div>
    </div>
    <div
      data-testid="piechart"
      style="width: 20rem; height: 20rem; border-radius: 50%; position: fixed; top: 100px; right: 10vw;"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-pfhr9p-MuiTypography-root"
      >
        <b>
          0.00
          %
        </b>
      </h4>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
